generator/template.go

```diff
@@ -576,13 +578,13 @@ func mapWellKnownType(protoType string) string {
 func mapScalaType(protoType string) string {
 	switch protoType {
 	case "uint64", "sint64", "int64", "fixed64", "sfixed64", "string":
-		return "string"
+		return "string | null"
 	case "float", "double", "int32", "sint32", "uint32", "fixed32", "sfixed32":
-		return "number"
+		return "number | null"
 	case "bool":
-		return "boolean"
+		return "boolean | null"
 	case "bytes":
-		return "Uint8Array"
+		return "Uint8Array | null"
```

Comment:
I would not expect this change to be necessary. As far as I know, we haven't seen that `null` is being serialized for fields that aren't of `google.protobuf._____Value` types. My understanding of this function is that it's for building the types for proto fields with types like `string` and `bool` and `int32`, which AFAIK shouldn't ever be serializing `null`, right?

Reply:
Oh maybe you're right. `google.protobuf._____Value` is used for optional params, isn't it? So if a param is type `string`, it should never be unset...?

I was thinking that a param of type `string` or type `number` could be potentially be unset but just hasn't happened in our codebase yet so making this change now would prevent us from running into this issue again in the future.

Reply:
Ah, I see. Yeah, if a field is `string` or `number`, then my understanding is that it always must be serialized as something (and if it's not set, it's serialized as the zero value of `""` or `0`).

So it should only be an issue with the well-known types. And there, I had just guessed that they serialize as `undefined` when they're not set, but apparently it's `null` instead. Yay!